runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleExtensions.kt

```diff
@@ -22,12 +22,27 @@ public fun <T : Any> SerialModule.getContextual(value: T): KSerializer<T>? {
     return getContextual(value::class)?.cast()
 }
 
+/**
+ * Attempts to retrieve a serializer from the current module and, if not found, fallbacks to [serializer] method
+ */
+@OptIn(UnsafeSerializationApi::class)
+public inline fun <reified T : Any> SerialModule.getContextualOrDefault(): KSerializer<T> =
+    // Even though serializer(KType) also invokes serializerOrNull, it is a significant performance optimization
+    // TODO replace with serializer(typeOf<T>()) when intrinsics are here
+    getContextual(T::class) ?: T::class.serializerOrNull() ?: serializer(typeOf<T>()).cast()
+
 /**
  * Returns a serializer associated with [klass], or a [default one][KClass.serializer].
  *
  * @throws SerializationException if serializer can't be found.
  */
 @ImplicitReflectionSerializer
+@OptIn(UnsafeSerializationApi::class)
+@Deprecated(
+    level = DeprecationLevel.WARNING,
+    message = "This method is deprecated for removal. Please use reified getContextualOrDefault<T>() instead",
+    replaceWith = ReplaceWith("getContextual(klass) ?: klass.serializer()")
+)
 public fun <T : Any> SerialModule.getContextualOrDefault(klass: KClass<T>): KSerializer<T> =
```

Comment:
Btw, I think we should add `getContextualOrDefault(type: KType)` -- for cases when full type information is present, but refinement is not possible (analog of `serializer<T>() / serializer(type)` pair)